frontend/services/entities/hosts.ts

```diff
@@ -202,6 +208,12 @@ export default {
     const label = getLabel(selectedLabels);
     const sortParams = getSortParams(sortBy);
 
+    // ensure macos_settings filter is always applied in
+    // conjuction with a team_id, 0 (no teams) by default
+    if (macSettingsStatus) {
```

Comment:
Does this null-ish coalescing logic need to be applied to any other endpoints? Maybe we should create a helper function that reconciles the team id param? I think we might also want this for team id and disk encryption status filter.

Reply:
Good idea – I'll create that follow-up issue

Reply:
@gillespi314 it's only 3 lines with minimal custom logic, do you think that warrants a helper function?

Reply:
Makes sense

Reply:
@gillespi314 how is this: https://github.com/fleetdm/fleet/pull/10472